Extension/src/common.ts

```diff
@@ -36,6 +37,61 @@ export function getRawPackageJson(): any {
     return rawPackageJson;
 }
 
+export function getRawTasksJson(): Promise<any> {
+    const path: string = getTasksJsonPath();
+    if (!path) {
+        return undefined;
+    }
+    return new Promise<any>((resolve, reject) => {
+        fs.exists(path, exists => {
+            if (!exists) {
+                return resolve({});
+            }
+            const fileContents: Buffer = fs.readFileSync(path);
+            let rawTasks: any = {};
+            try {
+                rawTasks = JSON.parse(fileContents.toString()); 
+            } catch (error) {
+            }
+            resolve(rawTasks);
+        });
+    });
+}
+
+export async function ensureBuildTaskExists(taskName: string): Promise<void> {
+    let rawTasksJson: any = await getRawTasksJson();
+
+    // Ensure that the task exists in the user's task.json. Task will not be found otherwise.
+    if (!rawTasksJson.tasks) {
+        rawTasksJson.tasks = new Array();
+    }
+    // Find or create the task which should be created based on the selected "debug configuration".
+    let selectedTask: vscode.Task = rawTasksJson.tasks.find(task => {
+        return task.label && task.label === task;
+    });
+    if (selectedTask) {
+        return;
+    }
+
+    const buildTasks: vscode.Task[] = await getBuildTasks();
+    selectedTask = buildTasks.find(task => task.name === taskName);
+    assert(selectedTask);
+
+    let definition: vscode.TaskDefinition = selectedTask.definition as vscode.TaskDefinition;
+    if (definition && definition.compilerPath) {
+         // TODO: add desired properties to empty object, don't delete.
+        delete definition.compilerPath;
+     }
+    rawTasksJson.tasks.push(selectedTask.definition);
+    // TODO: It's dangerous to overwrite this file. We could be wiping out comments.
+    await writeFileText(getTasksJsonPath(), JSON.stringify(rawTasksJson, null, 2));
```

Comment:
Is there a way we can use the vscode API to write to this file?  I would like to avoid losing existing comments in the file and it might be a bit of work to parse them out ourselves.  Perhaps the jsonc parser can help us out, but I'm unfamiliar with the API.

Reply:
I'm not sure! One possible solution is to invoke the "Configure Tasks" command, however it will pop up a quickpick menu for the user to select, and as far as I know the command itself doesn't take any arguments (gross). Otherwise the VS Code API does not expose any means as far as my research goes.

Another idea is to look at how VS Code does it to derive some possible solutions.